src/nylas-connection.ts

```diff
@@ -51,66 +62,60 @@ export default class NylasConnection {
     this.clientId = clientId;
   }
 
-  requestOptions(options: { [key: string]: any }) {
-    if (!options) {
-      options = {};
-    }
-    options = { ...options };
-    if (!options.method) {
-      options.method = 'GET';
-    }
-    if (options.path) {
-      if (!options.url) {
-        options.url = `${config.apiServer}${options.path}`;
-      }
-    }
-    if (!options.formData) {
-      if (!options.body) {
-        options.body = {};
-      }
-    }
-    if (options.json == null) {
-      options.json = true;
+  requestOptions(options: RequestOptions) {
+    const newOptions: CoreOptions & UrlOptions = {
+      method: options.method || 'GET',
+      url: `${config.apiServer}${options.path}`,
+      headers: options.headers || {},
+      json: options.json == null ? true : options.json,
```

Comment:
```suggestion
      json: options.json || true,
```
? don't know for sure if json=undefined (or any falsy value) makes any sense here

Reply:
This will not let to set `json: false`

Reply:
true is default == null checks if it's passed

Reply:
👍